### PR TITLE
Now users can create custom shortcodes and grouped shortcodes

### PR DESCRIPTION
**Pull request by jatinkatyal13**

### CODE REVIEW

File: src/db/models.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ export interface URLAttributes {
 
 export const URLs = db.define<URLAttributes, URLAttributes>('url', {
   code: {
-    type: Sequelize.INTEGER,
+    type: Sequelize.BIGINT,
```
Comment (jatinkatyal13) [inline src/db/models.ts]: @championswimmer When creating codes for characters of length 7-9 integers where greater than 2147483647 and hence I guess we'll need BIGINT

Comment (championswimmer) [reply src/db/models.ts]: +1

File: src/controllers/urls.ts [hunk 2]
```diff
@@ -28,13 +28,59 @@ export const createUrl = async (
       }
       const groupCode = splitShortCode[0]
 
-      const group = await Groups.findCreateFind({
-        where: { prefix: groupCode },
+      var group: GroupAttributes | null
```
Comment (championswimmer) [inline src/controllers/urls.ts]: no `var` only let and const

File: src/controllers/urls.ts [hunk 3]
```diff
@@ -28,13 +28,59 @@ export const createUrl = async (
...
+            ownerId: user.id,
+          },
+        })
+      }
+      if (!group) {
+        throw new Error('Could not find Group Code')
+      }
+
+      const opts = optsFromGroupedShortcode(group, splitShortCode[1])
+      const [url, urlCreated] = await URLs.findCreateFind({
```
Comment (championswimmer) [inline src/controllers/urls.ts]: `URLs.findCreateFind` is repeated everywhere. We do not need that. We just want to have `opts` separately created, the code after that is common

**Pull request closed without merging**